### PR TITLE
[luci-interpreter]Add SpaceToDepth kernels on luci-interpreter

### PR DESCRIPTION
**Pull request by struss**
This commit add `SpaceToDepth` operation kernels on luci-interpreter.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/SpaceToDepth.test.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  SpaceToDepth kernel(&input_tensor, &output_tensor, params);
+  kernel.configure();
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<float>(output_tensor),
+              ElementsAreArray(ArrayFloatNear(output_data)));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray(output_shape));
+}
+
+TEST(SpaceToDepthTest, Uint8)
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/SpaceToDepth.test.cpp:51]: `SpaceToDepth` does not care about quantization. Maybe TYPED_TEST instead?

Comment (struss) [reply compiler/luci-interpreter/src/kernels/SpaceToDepth.test.cpp:51]: Okay I will change it!

File: compiler/luci-interpreter/src/kernels/SpaceToDepth.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  assert(_output->element_type() == DataType::FLOAT32 || _output->element_type() == DataType::U8 ||
+         _output->element_type() == DataType::S8 || _output->element_type() == DataType::S32 ||
+         _output->element_type() == DataType::S64);
+  assert(_input->element_type() == _output->element_type());
+
+  const int block_size = params().block_size;
+  const int input_height = _input->shape().dim(1);
+  const int input_width = _input->shape().dim(2);
+  int output_height = input_height / block_size;
+  int output_width = input_width / block_size;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/SpaceToDepth.cpp:43]: (nitpick) I would use `int32_t` for values related to dimension size, just for consistency.

File: compiler/luci-interpreter/src/kernels/SpaceToDepth.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_KERNELS_SPACETODEPTH_H
+#define LUCI_INTERPRETER_KERNELS_SPACETODEPTH_H
+
+#include "core/Kernel.h"
+#include "core/KernelParams.h"
+
+#include <cstdint>
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/SpaceToDepth.h:23]: Looks unused

File: compiler/luci-interpreter/src/kernels/SpaceToDepth.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/SpaceToDepth.cpp:43]: okay, I will fix it.

File: compiler/luci-interpreter/src/kernels/SpaceToDepth.h (hunk 3 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/SpaceToDepth.h:23]: oh, yes. I will remove it.

**Pull request merged**